### ISSUE TITLE
[.NET Core] Unhandled exception stacktrace does not include source code location (file path and line number)

### ISSUE THREAD

**Issue by jeremymeng** [dotnet/cli#1706]
## Steps to reproduce

dotnet new
dotnet restore

add a line to throw exception 

``` csharp
            throw new Exception("Can we see the source code location?");
```
## Expected  behavior

Debug build, the stack trace should include source code locations for better diagnosing. Full .NET framework does that.

```
Unhandled Exception: System.Exception: can we see the source code line number?
   at ConsoleApplication.Program.Main(String[] args) in E:\temp\hello\Program.cs:line 10
```
## Actual behavior

Unhandled Exception: System.Exception: can we see the source code line number?
   at ConsoleApplication.Program.Main(String[] args)
## Environment data

`dotnet --version` output:

.NET Command Line Tools (1.0.0-beta-001632)

Product Information:
 Version:     1.0.0-beta-001632
 Commit Sha:  N/A

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 Runtime Id:  ubuntu.14.04-x64

**Comment by jeremymeng** [comment]
Not sure if this is the right repo.  Feel free to move.

**Comment by jeremymeng** [comment]
@davidfowl I remembered you mentioned that this feature was done but not merged for DNX.  Do you know any update?

/cc @piotrpMSFT

**Comment by iangcarroll** [comment]
This is pretty inconvenient when trying to debug something.

**Comment by piotrpMSFT** [comment]
I just followed up with @mikem8361 on this issue and learned that the underlying problem has been fixed in the 1.1 version of CoreFx. To get line numbers, update your `Microsoft.NetCore.App` reference to version `1.1.0`. You will need to include the CoreFx nightly feed in your NuGet configuration to make this work. The nightly feed URL can be found here: https://github.com/dotnet/corefx. I'll include a NuGet.config file that works at the time of this post:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
    <clear />
    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
    <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

And, finally, some output from an app that throws in `Main`:

```
➜  r1706 dotnet run
This app throws an exception, but on what line?

Unhandled Exception: System.Exception: Where am I?
   at Program.Main(String[] args) in /Users/piotrp/code/repro/r1706/Program.cs:line 9
```

/CC @livarcocc @jgoshi @krwq FYI
